src/lib.rs

```diff
@@ -135,23 +135,21 @@ where
     }
 
     pub fn check_data_ready(&mut self) -> Result<(), Error<E>> {
-        let mut retries = 10;
+        let mut retries = 20;
```

Comment:
@patrickelectric @joaoantoniocardoso 
Here is something I was concerned about:
For each register reading, like Data_Ready ( ST1 ), it takes 495 us.
However, Data_Ready will require the sample frequency to trigger.

For now, I've left out the thread::sleep and used 20 direct attempts.
For 200 Hz, it's actually working fine, with double the necessary attempts just to guarantee it.

For other modes, a different approach would be required. I was wondering if a watchdog/sleep approach might be more efficient for threads as well ( as it can allow other tasks to run while waiting (?) ).